### PR TITLE
TEST: Test Network for Amoy

### PR DESCRIPTION
**Pull request by Rampex1**
Linear issue:
- https://linear.app/streamingfast/issue/BLO-494/battlefield-amoy-tests

Tests:
- Validate unmergeable system tx (block 16)
- Validate combine system tx (block 35248)

### CODE REVIEW

**Comment by Rampex1** [comment]
1. Error message fixed
2. The reason I am using fh2.3 is that when I run it with fh3.0, I get the following error message: 
Error: You specified testing with fh3.0 but Firehose block version is 3 while fh3.0 expect version 4, it seems your geth instance is not running with the correct geth/Firehose version

I tried `make bor` on the branch `release/polygon-2.x-fh3.0` as well as `go install ./cmd/geth` on the branch `release/geth-1.x-fh3.0` but I keep getting the error above. Is it possible that the Amoy testnet is using the older version of the blocks?

File: test/lib/assertions.ts [hunk 1]
```diff
@@ -316,6 +316,10 @@ function assertTrxOrdinals(
   trace: TransactionTrace,
   blockNumber: number,
 ) {
+  // Skip ordinal uniqueness check for Amoy network
+  if (isNetwork && isNetwork("amoy")) {
+    return
+  }
```
Comment (maoueh) [inline test/lib/assertions.ts:322]: Relevant only in 2.3 context correct?

Comment (maoueh) [reply test/lib/assertions.ts:322]: I think we should probably be really precise and only skip for specific tests even.

File: README.md [hunk 2]
```diff
@@ -33,17 +33,18 @@ as for the chain your are testing against. See below for chain details and how y
...
+| Sei                     | `./scripts/run_firehose_sei.sh parallel`                                                  | `pnpm test:fh3.0:sei-dev`            | The `parallel` tag refers to transaction execution algorithm, test both   |
+| BNB                     | Docker miner: `./scripts/bnb/up.sh`, then `./scripts/run_firehose_bnb.sh`                 | `pnpm test:fh3.0:bnb-dev`            | None                                                                      |
+| Polygon (fh 2.3)        | Docker miner: `./scripts/polygon/up.sh 2.3`, then `./scripts/run_firehose_polygon.sh 2.3` | `pnpm test:fh2.3:polygon-dev`        | None                                                                      |
+| Polygon (fh 3.0)        | Docker miner: `./scripts/polygon/up.sh 3.0`, then `./scripts/run_firehose_polygon.sh 3.0` | `pnpm test:fh3.0:polygon-dev`        | None                                                                      |
+| Amoy                    | `./scripts/run_firehose_polygon.sh 3.0 amoy`                                              | `pnpm test:fh2.3:amoy --grep "Amoy"` | None                                                                      |
```
Comment (maoueh) [inline README.md:47]: This 2.3 needs to be updated here too right?

File: test/amoy.test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { expect } from "chai"
+import { fetchFirehoseBlock } from "./lib/firehose"
+import { isNetwork } from "./lib/network"
+
+if (isNetwork("amoy")) {
+  describe("Amoy", function (){
```
Comment (maoueh) [inline test/amoy.test.ts:6]: Do we have `Polygon` tests already? Would probably be good to actually merge it under Polygon?

File: test/global.ts [hunk 4]
```diff
@@ -12,6 +12,9 @@ import {
   Suicidal__factory,
   SuicideOnConstructor__factory,
   Transfers__factory,
+  DelegateToEmptyContract__factory,
+  SuicideContractAsBeneficiary__factory,
+  SuicideContractAsBeneficiarySameTrx__factory,
```
Comment (maoueh) [inline test/global.ts:17]: Thanks!

**Review by maoueh — CHANGES_REQUESTED** [review]

File: test/lib/assertions.ts (hunk 1 above)
Comment (Rampex1) [reply test/lib/assertions.ts]: This specifically happens for the test `Validate combine system transaction (block 35248)` where the error message displays that Ordinal 2 has been seen 2 times throughout transaction. It might be an edge case given this is dealing with system tx... not too sure.

For now, I have implemented a fix to skip ordinal check for that test specifically, but let me know if this ordinal duplication is not normal and should be investigated further.

**Review by maoueh — APPROVED** [review]

File: test/lib/assertions.ts (hunk 1 above)
Comment (maoueh) [reply test/lib/assertions.ts]: If that happen in fh3.0, I would say it's a problem. If that happen in `fh2.3`, we can skip it. That's why I wanted to have a good precision.

Let's discuss this tomorrow.
